barretenberg/cpp/src/barretenberg/vm2/simulation/poseidon2.cpp

```diff
@@ -54,4 +57,69 @@ std::array<FF, 4> Poseidon2::permutation(const std::array<FF, 4>& input)
     return output;
 }
 
+void Poseidon2::permutation(MemoryInterface& memory, MemoryAddress src_address, MemoryAddress dst_address)
+{
+
+    uint32_t execution_clk = execution_id_manager.get_execution_id();
+    uint32_t space_id = memory.get_space_id();
+
+    auto zero = MemoryValue::from<FF>(0);
+    std::array<MemoryValue, 4> input = { zero, zero, zero, zero };
```

Comment:
![zero](https://media0.giphy.com/media/v1.Y2lkPTc5MGI3NjExbmZsYXI5OG5taHl3bnQ5dHQwNjExaGR3amo1NzhxMTNxcm0zd3Q2byZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/l2JdYkTPVG9gRbvhK/giphy.gif)

Reply:
haha you suggested this one! This minimises the number of calls to the `MemoryValue` constructor 😉

Reply:
replaced with `input.fill(MemoryValue::from<FF>(0));`